Product list

obliviusm opened the pull request:
https://github.com/marchi-martius/FreemarketRn/issues/17

obliviusm committed:
static unstyled list

obliviusm committed:
add styling to list

obliviusm committed:
fix tests

obliviusm committed:
add snapshot tests for product list

obliviusm committed:
fix some lint

obliviusm commented:
![screenshot_1494631846](https://cloud.githubusercontent.com/assets/1386397/26020190/5d6ad6bc-3784-11e7-92fb-25d064204fa4.png)

obliviusm commented:
WIP. only UI yet.
working on server-redux thing

obliviusm committed:
fetch products to display

obliviusm committed:
use flat list

obliviusm committed:
fix tests

obliviusm committed:
add saga product test

obliviusm committed:
add test for product reducer

obliviusm committed:
fix lint

marchi-martius commented on app/components/CategoryOverviewView.js:
```diff
@@ -3,10 +3,12 @@ import PropTypes from 'prop-types';
 import { View } from 'react-native';
 
 import CategoryHeaderAndroid from './CategoryHeader.android';
+import ProductList from '../containers/ProductList';
```
Please avoid such hierarchy (pulling containers from other's container's component - view).
We should move all `ProductList` mapping to `CategoryOverview` and pass props downstream.

obliviusm replied:
@marchi-martius I see your point. But now I think that `CategoryOverviewView` doesn't give any UI. So it's not a component. We just need it to hold next containers. So I would rather move mapping below `CategoryOverviewView`, not above.

marchi-martius commented on app/components/__tests__/CategoryOverviewView.js:
```diff
@@ -1,17 +1,22 @@
-      platform="android"
-      onHeaderIconClicked={() => {}}
-    />,
+    <Provider store={store}>
```
My suggestion above will simplify such cases. BTW components should know nothing about store.

marchi-martius commented on app/containers/ProductList.js:
```diff
@@ -0,0 +1,14 @@
+  loadProducts: () => dispatch(requestProducts()),
+});
+
+export default connect(mapStateToProps, mapDispatchToProps)(ProductListView);
```
Take a look at my comments above about this container.

marchi-martius commented on app/sagas/products.js:
```diff
@@ -0,0 +1,22 @@
+  } catch (e) {
+    yield put(productsError(e));
+  }
+}
```
We should extract code duplication for fetching API. I'll open an Issue for this.

marchi-martius replied:
#26

obliviusm committed:
simplify platform-specific code

obliviusm committed:
remove CategoryOverview

obliviusm committed:
add empty ios header for tests

obliviusm committed:
add RN resolver

obliviusm commented:
@marchi-martius please, take look
Removed `CategoryOverview` for now.
Simplified platform code by https://facebook.github.io/react-native/docs/platform-specific-code.html#platform-specific-extensions

obliviusm committed:
remove cache flag
